include/fiction/algorithms/physical_design/wiring_reduction.hpp

```diff
@@ -540,29 +528,76 @@ create_wiring_reduction_layout(const Lyt& lyt, const uint64_t x_offset = 0, cons
                 wiring_reduction_lyt.obstruct_connection(shifted_tile_a, new_coord);
                 wiring_reduction_lyt.obstruct_connection(shifted_tile_b, new_coord);
 
-                wiring_reduction_lyt.obstruct_coordinate(new_coord);
-                wiring_reduction_lyt.obstruct_coordinate({new_coord.x, new_coord.y, 1});
+                obstruct_coordinate(0, 0);
+            }
 
-                // special case:
+            if (const auto signals = lyt.incoming_data_flow(old_coord); (old_coord.z == 1) || (signals.size() == 2))
+            {
+                // Special cases (where the crossing can also be placed further to the left or top):
                 // +→=
                 // ↓ ↓
                 // =→&
                 //
-                // -> No crossing between coordinate to the left and coordinate above the gate
-                if (!(lyt.has_northern_incoming_signal({old_coord.x - 1, old_coord.y, old_coord.z}) &&
-                      lyt.has_western_incoming_signal({old_coord.x, old_coord.y - 1, old_coord.z})))
-                {
-                    return;
-                }
-                if (wiring_reduction_lyt.get_search_direction() == search_direction::HORIZONTAL)
-                {
-                    wiring_reduction_lyt.obstruct_connection({new_coord.x - 1, new_coord.y, new_coord.z},
-                                                             {new_coord.x, new_coord.y - 1, new_coord.z});
-                }
-                else
+                // Or:
+                //
+                // +→=
+                // ↓ ↓
+                // =→+
+                if (is_single_wire(1, 0) && is_single_wire(0, 1))
                 {
-                    wiring_reduction_lyt.obstruct_connection({new_coord.x, new_coord.y - 1, new_coord.z},
-                                                             {new_coord.x - 1, new_coord.y, new_coord.z});
+                    uint64_t i        = 1;
+                    bool     obstruct = false;
+
+                    while (true)
```

Comment:
Could this be `for (uint64_t i = 1; true; ++i)`? Would save you the declaration and the two increments.